projects/plugins/protect/src/js/routes/scan/index.jsx

```diff
@@ -183,33 +189,22 @@ const ScanPage = () => {
 
 	// retry fetching status if it is not available
 	useEffect( () => {
-		if ( ! statusIsFetching && 'unavailable' === status.status && ! scanIsUnavailable ) {
+		if ( ! statusIsFetching && SCAN_STATUS_UNAVAILABLE === status.status && ! scanIsUnavailable ) {
 			refreshStatus( true );
 		}
 	}, [ statusIsFetching, status.status, refreshStatus, scanIsUnavailable ] );
 
 	const renderSection = useMemo( () => {
-		// Error
-		if ( error || scanIsUnavailable ) {
-			return <ErrorSection errorMessage={ errorMessage } errorCode={ errorCode } />;
+		if ( scanInProgress ) {
+			return <ScanningSection currentProgress={ status.currentProgress } />;
 		}
 
-		// Scanning
-		const scanningStatuses = new Set( [ 'scheduled', 'scanning', 'optimistically_scanning' ] );
-		if ( scanningStatuses.has( status.status ) || ! lastChecked ) {
-			return <ScanningSection currentProgress={ status.currentProgress } />;
+		if ( scanError ) {
+			return <ErrorSection errorMessage={ scanError.message } errorCode={ scanError.code } />;
```

Comment:
Prioritize showing an in-progress scan, as if the scan is running, the error may be from the old/current results.